app/javascript/react/components/ThresholdConfigurator/ThresholdConfigurator.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useRef, useState } from "react";
+
+import * as S from "./ThresholdConfigurator.style";
+
+const calculateThumbPercentage = (
+  value: number,
+  min: number,
+  max: number
+): number => {
+  const percentage = (value - min) / (max - min);
+  return percentage;
+};
+
+const calculateThumbPosition = (
+  value: number,
+  min: number,
+  max: number,
+  width: number
+): number => {
+  const percentage = calculateThumbPercentage(value, min, max);
+  return percentage * width;
+};
+
+interface ThresholdsConfiguratorProps {
+  initialThresholds: number[];
```

Comment:
I am not a fan of this for, few reasons:
1. We are not safe using arrays, because we are never sure about the number of elements presented inside an array
2. When accessing the threshold in the code below, we **assumed** it should be index `0` as min and `4` as max
3. We are accessing those thresholds in code multiple times, which makes us more error-prone. Human factors here can badly influence it, and by mistake someday we can access it as `min` ex. `1` instead of `0`
4. To be safe, as long as we are sure the number of thresholds is finite we should apply another interface and then use `thresholds.minimum`

This is the approach I am trying to apply when using type-safe languages, please skip this if you know it's not the approach we should take 🤝

Reply:
+1 to that.
I think we used enum in iOS, didn't we?

Reply:
Yes, I agree, thanks for the thorough comment! :)
I wasn't really thinking about data structures intentionally as waiting for the API shape, although given how much code I produced it could've been done. 😅

Reply:
Wait, we agreed with Marta, that API won't send the values most probably. 😅